components/server/ee/src/user/stripe-service.ts

```diff
@@ -35,140 +34,151 @@ export class StripeService {
+        const attributionId = AttributionId.render({ kind: "team", teamId: team.id });
+        if (await this.findCustomerByAttributionId(attributionId)) {
             throw new Error(`A Stripe customer already exists for team '${team.id}'`);
         }
         // Create the customer in Stripe
         const userName = User.getName(user);
         const customer = await this.getStripe().customers.create({
             email: User.getPrimaryEmail(user),
             name: userName ? `${userName} (${team.name})` : team.name,
-            metadata: {
-                [ATTRIBUTION_ID_METADATA_KEY]: AttributionId.render({ kind: "team", teamId: team.id }),
-            },
+            metadata: { attributionId },
         });
         // Wait for the customer to show up in Stripe search results before proceeding
         let attempts = 0;
-        while (!(await this.findCustomerByTeamId(team.id))) {
+        while (!(await this.findCustomerByAttributionId(attributionId))) {
             await new Promise((resolve) => setTimeout(resolve, POLL_CREATED_CUSTOMER_INTERVAL_MS));
             if (++attempts > POLL_CREATED_CUSTOMER_MAX_ATTEMPTS) {
                 throw new Error(`Could not confirm Stripe customer creation for team '${team.id}'`);
             }
         }
-        return customer;
+        return customer.id;
     }
 
-    async setPreferredCurrencyForCustomer(customer: Stripe.Customer, currency: string): Promise<void> {
-        await this.getStripe().customers.update(customer.id, { metadata: { preferredCurrency: currency } });
+    async setPreferredCurrencyForCustomer(customerId: string, currency: string): Promise<void> {
+        await this.getStripe().customers.update(customerId, { metadata: { preferredCurrency: currency } });
     }
 
-    async setDefaultPaymentMethodForCustomer(customer: Stripe.Customer, setupIntentId: string): Promise<void> {
+    async setDefaultPaymentMethodForCustomer(customerId: string, setupIntentId: string): Promise<void> {
         const setupIntent = await this.getStripe().setupIntents.retrieve(setupIntentId);
         if (typeof setupIntent.payment_method !== "string") {
             throw new Error("The provided Stripe SetupIntent does not have a valid payment method attached");
         }
         // Attach the provided payment method to the customer
         await this.getStripe().paymentMethods.attach(setupIntent.payment_method, {
-            customer: customer.id,
+            customer: customerId,
         });
-        await this.getStripe().customers.update(customer.id, {
+        const paymentMethod = await this.getStripe().paymentMethods.retrieve(setupIntent.payment_method);
+        await this.getStripe().customers.update(customerId, {
             invoice_settings: { default_payment_method: setupIntent.payment_method },
+            ...(paymentMethod.billing_details.address?.country
+                ? { address: { line1: "", country: paymentMethod.billing_details.address?.country } }
```

Comment:
Do you know why `line1` is mandatory? Why don't you copy over teh full address here?

Reply:
`line1` isn't actually mandatory: https://stripe.com/docs/api/customers/create#create_customer-address-line1

However, it seems that Stripe.js's TypeScript bindings marked it as mandatory, so I provide an empty value (`undefined` or `null` were not allowed, but I believe the result is the same).

Also, the country by itself is already the full address we get from the upgrade flow (we only ask for country, not for a full billing address):

<img width="400" alt="Screenshot 2022-09-23 at 10 14 47" src="https://user-images.githubusercontent.com/599268/191919141-8ebb8388-03d3-423d-9083-a212aff37ad3.png">

However, I'm not exactly sure what happens if:

- You subscribe from a given country
- Then, you set a complete billing address (e.g. all lines)
- Then, you cancel
- Then, you subscribe from a different country

It's sort of an edge case, but should also be tested. Thanks for pointing it out!

Reply:
Okay, I've tested it -- if you cancel and re-subscribe from a different country, your billing address is entirely reset:

<img width="600" alt="Screenshot 2022-09-23 at 15 45 02" src="https://user-images.githubusercontent.com/599268/191974824-7521272e-c9ef-4b6e-b303-de821472265b.png">

I think that's a good thing.